File: Arrays/118-Pascals-Triangle/118-Pascals-Triangle-Brute Force.py
"""
Problem: 118. Pascal's Triangle
Approach: Brute Force - Calculate Each Element

Each element is calculated using the binomial coefficient:
C(r - 1, c - 1)

Time Complexity: O(n^3)
Space Complexity: O(1) auxiliary space
"""


class Solution:

    def get_element(self, r, c):
        n = r - 1
        k = c - 1

        result = 1

        for i in range(k):
            result = result * (n - i) // (i + 1)

        return result

    def generate(self, numRows):
        ans = []

        for r in range(1, numRows + 1):
            row = []

            for c in range(1, r + 1):
                row.append(self.get_element(r, c))

            ans.append(row)

        return ans

File: Arrays/118-Pascals-Triangle/JavaScript/118-Pascals-Triangle-Brute Force.js
/**
 * Problem: 118. Pascal's Triangle
 * Approach: Brute Force - Calculate Each Element
 *
 * Each element is calculated using the binomial coefficient:
 * C(r - 1, c - 1)
 *
 * Time Complexity: O(n^3)
 * Space Complexity: O(1) auxiliary space
 */

/**
 * @param {number} r
 * @param {number} c
 * @return {number}
 */

var get_element = function(r, c) {
    let n = r - 1;
    let k = c - 1;

    let result = 1;

    for (let i = 0; i < k; i++) {
        result = Math.floor(result * (n - i) / (i + 1));
    }

    return result;
};

/**
 * @param {number} numRows
 * @return {number[][]}
 */

var generate = function(numRows) {
    let ans = [];

    for (let r = 1; r <= numRows; r++) {
        let row = [];

        for (let c = 1; c <= r; c++) {
            row.push(get_element(r, c));
        }

        ans.push(row);
    }

    return ans;
};